governance/program/src/processor/process_deposit_governing_tokens.rs

```diff
@@ -68,11 +92,7 @@ pub fn process_deposit_governing_tokens(
 
     if token_owner_record_info.data_is_empty() {
         // Deposited tokens can only be withdrawn by the owner so let's make sure the owner signed the transaction
-        let governing_token_owner = get_spl_token_owner(governing_token_source_info)?;
-
-        if !(governing_token_owner == *governing_token_owner_info.key
```

Comment:
Yes, it's actually a common use case for `Membership` scenarios where the membership is given to users and the tokens transferred or minted directly into the `Realm` holding account.

For the initial deposit there is still a check if the beneficiary (`governing_token_owner`)  signed the transaction but I'm thinking it could also be removed or a flag added to make the check optional because an empty `TokenOwnerRecord` can be created anyway using `CreateTokenOwnerRecord` instruction.